src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -679,6 +688,31 @@ private static boolean isLambdaSingleParameterSurrounded(DetailAST ast) {
         return result;
     }
 
+    /**
+     *  Returns the direct LPAREN tokens children to a given QUESTION token which
+     *  contain an expression not a literal variable.
+     *
+     *  @param questionToken {@code DetailAST} question token to be checked
+     *  @return the direct children to the given question token which their types are LPAREN
+     *          tokens and not contain a literal inside the parentheses
+     */
+    private static List<DetailAST> getParenthesesChildrenAroundQuestion(DetailAST questionToken) {
+        final List<DetailAST> allChildren = new ArrayList<>();
+        DetailAST directChild = questionToken.getFirstChild();
+        while (directChild != null) {
+            allChildren.add(directChild);
+            directChild = directChild.getNextSibling();
+        }
+        final List<DetailAST> surroundedChildren = new ArrayList<>();
+        for (DetailAST child : allChildren) {
+            if (child.getType() == TokenTypes.LPAREN
+                    && !TokenUtil.isOfType(child.getNextSibling(), LITERALS)) {
+                surroundedChildren.add(child);
+            }
+        }
+        return surroundedChildren;
```

Comment:
Return it as `unmodifiableList` using `Collections.unmodifiableList(childern);`

Reply:
done